src/main/rx/srxl2.c

```diff
@@ -0,0 +1,610 @@
+static volatile uint32_t lastValidPacketTimestamp = 0;
+static volatile uint32_t lastIdleTimestamp = 0;
+
+static volatile uint8_t readBuffer[2 * SRXL2_MAX_PACKET_LENGTH];
+static volatile uint8_t readBufferIdx = 0;
+static volatile uint8_t readBufferReadIdx = 0;
+static uint8_t writeBuffer[SRXL2_MAX_PACKET_LENGTH];
+static uint8_t writeBufferIdx = 0;
+
+static serialPort_t *serialPort;
+
+static uint8_t busMasterDeviceId = 0xFF;
+static bool telemetryRequested = false;
+
+static uint8_t telemetryFrame[22];
+
+uint8_t globalResult = 0;
+bool srxl2ProcessHandshake(const Srxl2Header* header, const Srxl2HandshakeSubHeader* handshake)
+{
+    if (handshake->destinationDeviceId == Broadcast) {
+        DEBUG("broadcast handshake from %x\r\n", handshake->sourceDeviceId);
+        busMasterDeviceId = handshake->sourceDeviceId;
+
+        if (handshake->baudSupported == 1) {
+            serialSetBaudRate(serialPort, SRXL2_PORT_BAUDRATE_HIGH);
+            DEBUG("switching to %d baud\r\n", SRXL2_PORT_BAUDRATE_HIGH);
+        }
+
+        state = Running;
+
+        return true;
+    }
+
+
+    if (handshake->destinationDeviceId != ((FlightController << 4) | unitId)) {
+        return true;
+    }
+
+    DEBUG("FC handshake from %x\r\n", handshake->sourceDeviceId);
+
+    Srxl2HandshakeFrame response = {
+        .header = *header,
+        .payload = {
+            handshake->destinationDeviceId,
+            handshake->sourceDeviceId,
+            /* priority */ 10,
+            /* baudSupported*/ baudRate,
+            /* info */ 0,
+            U_ID_2
+        }
+    };
+
+    const uint16_t crc = crc16_ccitt_update(0, &response, sizeof(response) - 2);
+    response.crcHigh = (crc >> 8);
+    response.crcLow = (crc & 0xFF);
+
+    srxl2RxWriteData(&response, sizeof(response));
+
+    return true;
+}
+
+void srxl2ProcessChannelData(const Srxl2ChannelDataHeader* channelData, rxRuntimeConfig_t *rxRuntimeConfig) {
+    if (channelData->rssi >= 0) {
+        const int rssiPercent = channelData->rssi;
+        setRssi(scaleRange(rssiPercent, 0, 100, 0, RSSI_MAX_VALUE), RSSI_SOURCE_RX_PROTOCOL);
+    } else {
+        const int rssiDbm = channelData->rssi;
+        (void) rssiDbm;
+        // @todo come up with a scheme to scale power values to 0-1023 range
+        setRssi(RSSI_MAX_VALUE / 2, RSSI_SOURCE_RX_PROTOCOL);
+    }
+
+    if (channelData->rssi == 0) {
+        globalResult = RX_FRAME_FAILSAFE;
+    } else {
+        globalResult = RX_FRAME_COMPLETE;
+    }
+
+    const uint16_t *frameChannels = (const uint16_t *) (channelData + 1);
+
+    for(size_t i = 0; i<4; i++)
+    {
+        if (channelData->channelMask.u8[i]) {
+            for (size_t j = 0; j < 8; ++j) {
+                if (channelData->channelMask.u8[i]>> j & 1) {
+                    rxRuntimeConfig->channelData[i*8+j] = *frameChannels++;
+                }
+            }
+        }
+    }
+
+    // DEBUG("channel data: %d %d %x\r\n",
```

Comment:
Please don't leave commented out code in without an annotation - state when and under what circumstances it should be re-enabled, or remove it.